tests/integration/test_utils.py

```diff
@@ -198,17 +211,20 @@ def merge_resources(existing_resource, new_resource):
             existing_scopes.append(new_scope)
             
 def custom_json_merge(result, new_json):
-    new_resources = {resource_sorting_key(resource): resource for resource in new_json["resourceMetrics"]}
```

Comment:
This line (and few similar above) was main source of instability. Basically what it did was that it create Map and in case there was multiple resources in the same json with the same key, it did not merge them, but override them completely. So instead of creating Map I now create array of Tuples so not overriding anything